Physics/Dynamics/ProductionFormFactor.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2013, 2017 The ComPWA Team.
+// This file is part of the ComPWA framework, check
+// https://github.com/ComPWA/ComPWA/license.txt for details.
+
+#include <cmath>
+#include <iterator>
+#include <numeric>
+#include <vector>
+
+#include "Coupling.hpp"
+#include "Physics/HelicityFormalism/HelicityKinematics.hpp"
+#include "ProductionFormFactor.hpp"
+
+namespace ComPWA {
+namespace Physics {
+namespace Dynamics {
+
+ProductionFormFactor::ProductionFormFactor(std::string name,
+                                           std::shared_ptr<ComPWA::FitParameter> mass1,
+                                           std::shared_ptr<ComPWA::FitParameter> mass2,
+                                           std::shared_ptr<ComPWA::FitParameter> radius,
+                                           ComPWA::Spin orbitL,
+                                           FormFactorType ffType)
+    : AbstractDynamicalFunction(name),
+    Daughter1Mass(mass1), Daughter2Mass(mass2), MesonRadius(radius),
+    L(orbitL), FFType(ffType) {
+
+  LOG(TRACE) << "ProductionFormFactor::Factory() | Construction production "
+             << "formfactor of " << name << ".";
+}
+
+ProductionFormFactor::~ProductionFormFactor() {}
+
+std::complex<double> ProductionFormFactor::evaluate(const DataPoint &point,
```

Comment:
Here you would call the "undecorated" member's `evaluate()` function and multiply this class `dynamicalFunction()` function on top. Maybe even rename `dynamicalFunction()` here to something more meaningful.